src/http/response_body.rs

```diff
@@ -107,11 +107,11 @@ impl ResponseBody {
             return Ok(());
           }
 
-          if read > io_buf.len() {
-            return Err(io::Error::new(io::ErrorKind::Other, "buffer overflow"));
-          }
-
-          stream.write_all(&io_buf[..read])?;
+          stream.write_all(
+            io_buf
+              .get_mut(..read)
+              .ok_or(io::Error::new(io::ErrorKind::Other, "buffer overflow"))?,
```

Comment:
isnt there a ok_or_call_closure(|| io::Error::new(io::ErrorKind::Other, "buffer overflow")) variant? 
I am not sure the compiler knows that in the ok case that io::Error::new has no side effects and should be optimized out.